src/app/GitCommands/Settings/AppSettings.cs

```diff
@@ -1029,6 +1029,12 @@ public static LocalChangesAction CheckoutBranchAction
             set => SetEnum("checkoutbranchaction", value);
         }
 
+        public static bool CheckoutOtherBranchAfterReset
+        {
+            get => GetBool("checkoutotherbranchafterreset", true);
+            set => SetBool("checkoutotherbranchafterreset", value);
+        }
```

Comment:
Use `ISetting<bool>`, please, and update `AppSettingsTests`.

Reply:
I'm a bit rusty on rules when to use `ISetting<T>`... Isn't it used when a setting is shown in the Settings dialog?

Reply:
The rule is simple: Use `ISettings<T>` for every (new) AppSetting. :)